otherlibs/dune-site/src/plugins/plugins.ml

```diff
@@ -1,10 +1,16 @@
 open Dune_site.Private_
 module Data = Dune_site_plugins_data
 
+let meta_fn = "META"
+
 let readdir dirs =
+  let ( / ) = Filename.concat in
   List.concat
     (List.map
-       (fun dir -> Array.to_list (Sys.readdir dir))
+       (fun dir ->
```

Comment:
this can probably be a `List.filter_map` instead.

It's available since OCaml 4.08, so it should be safe to use inside dune (https://github.com/ocaml/ocaml/blob/8b6b7cfbf26ae50ae87d5849c827e609a1309c98/stdlib/list.mli#L195-L200)